go/multipooler/manager/manager.go

```diff
@@ -1328,3 +1342,44 @@ func (pm *MultiPoolerManager) Start(senv *servenv.ServEnv) {
 		pm.logger.Info("MultiPoolerManager gRPC services registered")
 	})
 }
+
+// WaitUntilReady blocks until the manager reaches Ready or Error state, or
+// the context is cancelled. Returns nil if Ready, or an error if Error state
+// or context cancelled. This should be called after Start() to ensure
+// initialization is complete before accepting RPC requests.
+//
+// Thread-safety: This method safely reads manager state by acquiring and
+// releasing pm.mu for each check. This prevents holding the lock during the
+// polling interval, allowing background goroutines to update state
+// without blocking.
+func (pm *MultiPoolerManager) WaitUntilReady(ctx context.Context) error {
+	ticker := time.NewTicker(50 * time.Millisecond)
```

Comment:
Polling here isn't my favorite. Could be cleaner to use a channel to signal when startup has either succeeded or failed?

I'll defer to your judgement but to me polling feels more complex, slower to react, and less idiomatic than receiving kind of a direct signal

Reply:
I've added `startupChan` for `setStateError` and `checkAndSetReady` to signal that they've executed